provider/src/main/kotlin/org/gradle/kotlin/dsl/NamedDomainObjectContainerExtensions.kt

```diff
@@ -46,7 +44,19 @@ inline operator fun <T : Any, C : NamedDomainObjectContainer<T>> C.invoke(
  */
 class NamedDomainObjectContainerScope<T : Any>(
     private val container: NamedDomainObjectContainer<T>
-) : NamedDomainObjectContainer<T> by container {
+) : NamedDomainObjectContainer<T> by container, PolymorphicDomainObjectContainer<T> {
+
+    override fun <U : T> create(name: String, type: Class<U>): U =
+        polymorphicDomainObjectContainer().create(name, type)
+
+    override fun <U : T> create(name: String, type: Class<U>, configuration: Action<in U>): U =
+        polymorphicDomainObjectContainer().create(name, type, configuration)
+
+    override fun <U : T> maybeCreate(name: String, type: Class<U>): U =
+        polymorphicDomainObjectContainer().maybeCreate(name, type)
+
+    override fun <U : T> containerWithType(type: Class<U>): NamedDomainObjectContainer<U> =
+        polymorphicDomainObjectContainer().containerWithType(type)
```

Comment:
Don't you want this to return a `NamedDomainObjectContainerScope`?

Reply:
A simple `NamedDomainObjectContainer` is enough, the scope for the returned container will be provided by https://github.com/gradle/kotlin-dsl/pull/873/files/ade1cc6b0cf0287ea074791a86e599df53bc2a4a#diff-3527cb19f2f40291861f41b1ffe878bfR33

```kotlin
container.containerWithType(...) {
    // scoped
}
// or
val other = container.containerWithType(...)
other {
    // scoped
}
```